### ISSUE TITLE
String foreign keys cannot be migrated in MySQL

### ISSUE THREAD

**Issue by bokub** [go-gorm/gorm#3866]
## GORM Playground Link

https://github.com/go-gorm/playground/pull/232

## Description

Hi again :wave: 
If I choose `string` for the type of an ID **which is also a foreign key**, the auto-migration creates  the foreing key as a `longtext`, which is not suitable for a foreign key in MySQL. The auto-migration throws an error.

More details in the playground PR

**Comment by jinzhu** [comment]
Hi, you can set a size here. https://github.com/go-gorm/playground/pull/232/files#diff-72487d6e423e37c0af7994baddca95c1e2acc7e7ca51b8d13d03caa8e4485a16R53

**Comment by bokub** [comment]
Yes I figured it out thanks to #3864 , but I spent a lot of time figuring out what was wrong in my code, while it was in fact a problem from GORM (in my opinion).

From what I understand, **regular string fields**  are created as `longtext`, but for **string fields that are the primary key**, GORM creates a `varchar` instead. It's **automatic**, the user doesn't have to add some special tag
 
Why should it be different for foreign keys? Especially if you want the foreign key to have the same type as the primary key
